src/Aspire.Cli/CliExecutionContext.cs

```diff
@@ -0,0 +1,10 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Cli;
+
+internal class CliExecutionContext(DirectoryInfo workingDirectory, DirectoryInfo hivesDirectory)
```

Comment:
We've been somewhat inconsistent with our detection of the current working directory. I'm replacing sloppy working directory detection with this injected context object as I go. I needed it for this PR - but as I see things I'm touching them up too.